File: go/internal/scoring/testdata/structural-v1/typescript/state_cosmetic_after.ts
export class CosmeticAfter {
  private total = 0;
  private updates = 0;

  private acceptState(delta: number): boolean {
    if (delta < 0) return false;
    return true;
  }
  private setState(delta: number): void {
    this.total += delta;
    if (this.total > 100) this.total = 100;
  }
  private markState(): void { this.updates++; }

  advance(delta: number): number {
    if (!this.acceptState(delta)) return this.total;
    this.setState(delta);
    this.markState();
    return this.total;
  }
}

File: go/internal/scoring/testdata/structural-v1/rust/state_cosmetic_after.rs
pub struct CosmeticAfter { total: i32, updates: i32 }

impl CosmeticAfter {
    fn accept_state(delta: i32) -> bool {
        if delta < 0 { return false; }
        true
    }
    fn set_state(&mut self, delta: i32) {
        self.total += delta;
        if self.total > 100 { self.total = 100; }
    }
    fn mark_state(&mut self) { self.updates += 1; }

    pub fn advance(&mut self, delta: i32) -> i32 {
        if !Self::accept_state(delta) { return self.total; }
        self.set_state(delta);
        self.mark_state();
        self.total
    }
}
